python/py-isce2/Portfile

```diff
@@ -0,0 +1,155 @@
+
+
+Other important notes
+----------------------
+
+1) GDAL is required for working with Radarsat2, Terrasar-X and Sentinel 1-A. Suggested variant of gdal is  +expat+geos+hdf5+netcdf+postgresql95+sqlite3
+
+2) One can use isce2 with other packages like SpiceyPy, sarpy etc. These will need to be installed from source separately.
+
+    "
+
+    depends_lib-append      port:curl \
+                            port:bzip2\
+                            port:openmotif\
+                            port:hdf5 \
+                            port:fftw-3\
+                            port:fftw-3-single\
+                            port:gdal \
+                            port:scons\
+                            port:py${python.version}-setuptools\
+                            port:py${python.version}-cython\
+                            port:py${python.version}-numpy \
+                            port:py${python.version}-scipy \
+                            port:py${python.version}-nose  \
+                            port:py${python.version}-gdal  \
+                            port:py${python.version}-h5py
+
+    require_active_variants port:curl ssl
+    require_active_variants port:gdal hdf5
+    require_active_variants port:gdal netcdf
+
+    post-patch {
+        ###Replace cython3 with full path
+        set cypath ${prefix}/bin/cython-${python.branch}
+        reinplace "s|cython3|${cypath}|g" ${worksrcpath}/SConstruct
+        reinplace "s|cython3|${cypath}|g" ${worksrcpath}/library/isceLib/pyx/SConscript
+        reinplace "s|cython3|${cypath}|g" ${worksrcpath}/contrib/splitSpectrum/pyx/SConscript
+
+        ###Replace /usr/bin/env/python in all python files
+        ###From Portfile of nodejs10
+        foreach f [rec_glob ${worksrcpath} *.py] {
+            reinplace -q "s|/usr/bin/env python3|/usr/bin/env python${python.branch}|g" ${f}
+        }
+    }
+
+    pre-configure {
+        xinstall -d ${workconfpath}
+        xinstall -d ${workbldpath}
+    }
+
+    configure {
+        #Check if compiler is set
+        if {![gcc_variant_isset]} {
+            default_variants-append +gcc7
+        }
+
+        #For gcc, set libstdc++
+        configure.cxx_stdlib libstdc++
+
+    	#Setup SConfigISCE and config dir here
+        set gccvar  [gcc_variant_name]
+        set gcclast [string index ${gccvar} end]
+        set cfgname ${workconfpath}/SConfigISCE
+
+        file copy -force ${filespath}/gcc_SconfigISCE ${cfgname}
+        reinplace "s|@@BLDPATH@@|${workbldpath}|g" ${cfgname}
+        reinplace "s|@@INSTALLPATH@@|${workinstallpath}|g" ${cfgname}
+        reinplace "s|@@PREFIX@@|${prefix}|g" ${cfgname}
+        reinplace "s|@@PYINCLUDE@@|${python.include}|g" ${cfgname}
+        reinplace "s|@@GCCLAST@@|${gcclast}|g" ${cfgname}
+    }
+
+
+    destroot {
+        xinstall -m 755 -d ${destroot}${python.pkgd}/isce
+        system "cp -rf ${workinstallpath}/* ${destroot}${python.pkgd}/isce"
```

Comment:
Can you not use `file copy` or `xinstall` recursively here instead of using a system command to copy the files?

Reply:
https://trac.macports.org/ticket/14139

I used the same pattern as the ones already in use at destroot stage by more than 100 ports. If there is an example, that you can point me to that does this recursively - will be happy to use the same mechanism.

Reply:
OK. No problem.

Reply:
Looks like simple "file copy" works for directories. It would be nice to have this documented. I have made this change.

Reply:
Submit a documentation pull request?